spec/spec_helper.rb

```diff
@@ -3,4 +3,19 @@
   SimpleCov.start
 end
 
-Dir[Rails.root.join("spec/shared/**/*.rb")].each { |f| require f }
\ No newline at end of file
+require 'manageiq-consumption'
+require 'money-rails/test_helpers'
+require 'money-rails'
+
+
+Dir[ManageIQ::Consumption::Engine.root.join("spec/support/**/*.rb")].each { |f| require f }
+
+require 'factory_girl'
+# in case we are running as an engine, the factories are located in the dummy app
+FactoryGirl.definition_file_paths << 'spec/factories'
+# also add factories from provider gems until miq codebase does not use any provider specific factories anymore
+Rails::Engine.subclasses.select { |e| e.name.starts_with?("ManageIQ::Consumption") }.each do |engine|
+  FactoryGirl.definition_file_paths << File.join(engine.root, 'spec', 'factories')
+end
```

Comment:
@sergio-ocon since tests from manageiq won't be loading your spec_helper, you need to tell them where you factories are located via a rails engine initializer:

```
diff --git a/lib/manageiq/consumption/engine.rb b/lib/manageiq/consumption/engine.rb
index f1048c4..1b98f8c 100644
--- a/lib/manageiq/consumption/engine.rb
+++ b/lib/manageiq/consumption/engine.rb
@@ -6,9 +6,13 @@ module ManageIQ
     class Engine < ::Rails::Engine
       isolate_namespace ManageIQ::Consumption

+      initializer "model_core.factories", :after => "factory_girl.set_factory_paths" do
+        FactoryGirl.definition_file_paths << File.expand_path('../../../spec/factories', __dir__) if defined?(FactoryGirl)
+        puts FactoryGirl.definition_file_paths.inspect
+      end
       def vmdb_plugin?
         true
       end
     end
   end
-end
\ No newline at end of file
+end
```